rio/subsystems/drivetrain.py

```diff
@@ -152,21 +201,51 @@ def setModuleState(self, newState):
         the "state" (steering and speed)
         of a module.
         """
-
         # TODO: Use optimization at some point
-        self.state = newState
+
+        optimizedState = kinematics.SwerveModuleState.optimize(
+            newState, self.getModuleState().angle
+        )
```

Comment:
The optimized angle may be just half the picture.

if we say we want to go from 90, to -89, it inverts as normal and tells us our heading angle is 91, but if we say we want to go from 180, to -179, it still inverts us but we run out of travel

i think this is why we see the 90 degree shift trying to drive the wheel around.